pkg/controllers/management/globaldns/globaldns_handler.go

```diff
@@ -176,6 +189,51 @@ func (n *GDController) generateNewIngressSpec(globaldns *v3.GlobalDns) *v1beta1.
 	}
 }
 
+func (n *GDController) getIngressClass(globalDNSProviderName string) (string, error) {
+	providerName, err := n.getGlobalDNSProviderName(globalDNSProviderName)
+	if err != nil {
+		return defaultIngressClass, err
+	}
+
+	provider, err := n.globalDNSProviderLister.Get(namespace.GlobalNamespace, providerName)
+	if err != nil && k8serrors.IsNotFound(err) {
+		logrus.Errorf("GlobalDNSController: Object Not found Error %v, while listing GlobalDNSProvider by name %v", err, providerName)
+		return defaultIngressClass, nil
+	}
+	if err != nil {
+		return defaultIngressClass, fmt.Errorf("GlobalDNSController: Error %v Listing GlobalDNSProvider by name %v", err, providerName)
+	}
+
+	var options map[string]string
+	if provider.Spec.Route53ProviderConfig != nil {
+		options = provider.Spec.Route53ProviderConfig.AdditionalOptions
+	} else if provider.Spec.CloudflareProviderConfig != nil {
+		options = provider.Spec.CloudflareProviderConfig.AdditionalOptions
+	} else if provider.Spec.AlidnsProviderConfig != nil {
+		options = provider.Spec.AlidnsProviderConfig.AdditionalOptions
+	}
+
+	if options != nil {
+		ingressClassAnnotation, ok := options[annotationFilter]
+		if ok {
+			prefix := annotationIngressClass + "="
+			if strings.HasPrefix(ingressClassAnnotation, prefix) {
+				return strings.TrimPrefix(ingressClassAnnotation, prefix), nil
+			}
+		}
+	}
+	return defaultIngressClass, nil
```

Comment:
question, as per [this doc](https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/public-private-route53.md) linked in the issue, the annotation-filter's value can only have external-dns or internal-dns as the ingress class, so if the annotation is not present, why do we have to return rancher-external-dns as the ingress class? does it get picked as external or internal?

Reply:
The annotation-filter's value can be any thing that user wants to put. It just means that external-dns will only choose the ingresses that carry this annotation

Now we  add this annotation `kubernetes.io/ingress.class: rancher-external-dns` to all the dummy ingresses we create as part of GlobalDNS controller so that our Rancher's nginx ingress-controller does not program it by mistake.
So now if user provides an annotation filter for the provider, we place that or if user does not we default to `kubernetes.io/ingress.class: rancher-external-dns`